design/open/delta-diff.md

```diff
@@ -0,0 +1,186 @@
+The diff algorithm:
+1. Run the Delta [HISTORY command](https://docs.delta.io/latest/delta-utility.html#history-schema) on both table paths.
+2. Traverse through the [returned "commitInfo" entry vector ](https://github.com/delta-io/delta-rs/blob/d444cdf7588503c1ebfceec90d4d2fadbd50a703/rust/src/delta.rs#L910)
+starting from the **last** version of each history vector: 
+    1. While the returned entry versions **aren't** equal OR no more versions available:
+        1. If the bigger version is the _"left"'s_ version, add the entry to the returned history list.
+        2. Traverse one version back of the bigger version entry's history.
+    2. While the versions > 0 OR no more versions available: 
+        1. Create a hash for the entry based on fields: `timestamp`, `operation`, `operationParameters` , and `operationMetrics` values.
+        2. Compare the hashes of the versions.
+        3. If they **aren't equal**, add the "left"'s entry to the returned history list, else, break and **return the history list**.
+        4. Traverse one version back in both vectors.
+3. Return the history list.
+
+### Authentication
+
+The `delta-rs` package generates an S3 client which will be used to communicate back to lakeFS (through the S3 gateway).  
+In order for the S3 client to communicate with lakeFS (or S3 in general) it needs to pass an AWS Access Key Id and Secret access key.  
+Since applicative credentials are not obligatory, the users that sent the request for a diff might not have such, and even if they have,
+they cannot send them through the GUI (which is the UI we chose to implement this feature for the MVP).
+To overcome this scenario, we'll use special diff credentials as follows:
+1. User makes a Delta Diff request.
+2. The DiffService checks if the user has "diff credentials" in the DB:
+    1. If there are such credentials, it will use them.
+    2. If there aren't such, it will generate the credentials and save them: `{AKIA: DIFF-<>, SAK: <>}`. The `DIFF` prefix will be used to identify "diff credentials".
+3. The DiffService will pass the credentials to the Delta Diff plugin during the call as part of the `gRPC` call.
+
+### API
+
+- GET `/repositories/repo/{repo}/otf/refs/{left_ref}/diff/{right_ref}?type={diff_type}`
+    - Tagged as experimental
+    - **Response**:  
+        The response includes an array of operations from different versions of the specified table format.
+        It has a general structure that enables formatting the different table format operation structs.
+      - version:
+        - type: string
+        - description: the version/snapshot/transaction id of the operation.
+      - timestamp (epoch):
+        - type: long
+        - description: operation's timestamp.
+      - operation:
+        - type: string
+        - description: operation's name.
+      - operationContent:
+        - type: map
+        - description: an operation content specific to the table format implemented.
```

Comment:
Just verify - what types the map will hold? we need to serialize any json object?